### PR TITLE
Jmpi 174 flag candidates for review on recompute

### PR DESCRIPTION
**Pull request by ASomaroo**
JMPI-174: flag for review with reason

### CODE REVIEW

**Comment by ASomaroo** [comment]
Looking into the way errors are handled and propagated, we will probably add the error handling for the updateMatchingPatientRecordScoreForGoldenRecord flow to this commit.

**Review by MatthewErispe — COMMENTED** [review]
please see my comments

File: JeMPI_Apps/JeMPI_Linker/src/main/java/org/jembi/jempi/linker/BackEnd.java [hunk 1]
```diff
@@ -466,4 +485,39 @@ public record EventLinkPatientToGidSyncRsp(
...
+         return notificationCandidates;
+      } catch (Exception e) {
+         LOGGER.error(e.getMessage());
+         return new ArrayList<>();
+      }
+   }
+
+   private static boolean isWithinThreshold(final float score) {
+      float minThreshold = AppConfig.BACK_END_MATCH_THRESHOLD - 0.1f;
+      float maxThreshold = AppConfig.BACK_END_MATCH_THRESHOLD + 0.1f;
```
Comment (marrouchi) [inline JeMPI_Apps/JeMPI_Linker/src/main/java/org/jembi/jempi/linker/BackEnd.java:518]: How about we add "0.1f" in AppConfig as well ?

File: JeMPI_Apps/JeMPI_Linker/src/main/java/org/jembi/jempi/linker/BackEnd.java [hunk 2]
```diff
@@ -466,4 +485,39 @@ public record EventLinkPatientToGidSyncRsp(
...
 
+   public static ArrayList<Notification.MatchData> getCandidates(final PatientRecord patientRecord) throws RuntimeException {
+
+      try {
+         List<GoldenRecord> candidateGoldenRecords =
+               libMPI.getCandidates(patientRecord.demographicData(), AppConfig.BACK_END_DETERMINISTIC);
+         List<WorkCandidate> allCandidateScores =
+               candidateGoldenRecords.parallelStream()
+                                     .unordered()
+                                     .map(candidate -> new WorkCandidate(candidate,
```
Comment (marrouchi) [inline JeMPI_Apps/JeMPI_Linker/src/main/java/org/jembi/jempi/linker/BackEnd.java:496]: maybe we could chain a ".filter()" function after this to get candidates with scores within threshold and avoid having a "for" loop afterwards. Does it sound right/better ?

File: JeMPI_Apps/JeMPI_Linker/src/main/java/org/jembi/jempi/linker/BackEnd.java [hunk 3]
```diff
@@ -466,4 +485,39 @@ public record EventLinkPatientToGidSyncRsp(
          LinkInfo linkInfo) implements EventResponse {
    }
 
+   public static ArrayList<Notification.MatchData> getCandidates(final PatientRecord patientRecord) throws RuntimeException {
```
Comment (marrouchi) [inline JeMPI_Apps/JeMPI_Linker/src/main/java/org/jembi/jempi/linker/BackEnd.java:488]: The function returns candidate match data notifications, so maybe we should use more appropriate name ? WDYT ?

**Review by marrouchi — CHANGES_REQUESTED** [review]

**Review by marrouchi — APPROVED** [review]
👍

**Pull request merged**